ldap/servers/plugins/replication/repl5_agmt.c

```diff
@@ -358,8 +364,33 @@ agmt_new_from_entry(Slapi_Entry *e)
         ra->timeout = timeout;
     }
 
+    /* DN of entry at root of replicated area */
+    tmpstr = slapi_entry_attr_get_charptr(e, type_nsds5ReplicaRoot);
+    if (NULL != tmpstr) {
+        ra->replarea = slapi_sdn_new_dn_passin(tmpstr);
+
+        /* now that we set the repl area, when can bump our agmt count */
+        if ((replica = replica_get_replica_from_dn(ra->replarea))) {
+            replica_incr_agmt_count(replica);
+        }
+        be = slapi_be_select(ra->replarea);
+    }
+    if (!be) {
+        slapi_log_err(SLAPI_LOG_ERR, repl_plugin_name,
+                      "agmt_new_from_entry - Failed to get backend for agreement %s on replicated suffix %s).\n",
+                      slapi_entry_get_dn(e), tmpstr ? tmpstr : "<NULL>");
+        goto loser;
+    }
+    use_lmdb = dblayer_is_lmdb(be);
```

Comment:
Minor nitpick. I think it'll be a bit more readable if `use_lmdb = dblayer_is_lmdb(be);` line will be right before `ra->flowControlWindow = use_lmdb ? LMDB_DEFAULT_FLOWCONTROL_WINDOW : DEFAULT_FLOWCONTROL_WINDOW;` line